### PR TITLE
process blocks

### PR DESCRIPTION
**Pull request by miiu96**
## Reasoning behind the pull request
- Process blocks

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

File: node/chainSimulator/components/testOnlyProcessingNode.go [hunk 1]
```diff
@@ -151,11 +151,6 @@ func NewTestOnlyProcessingNode(args ArgsTestOnlyProcessingNode) (*testOnlyProces
 	if err != nil {
 		return nil, err
 	}
-	err = instance.createTransactionLogProcessor()
```
Comment (iulianpascalau) [inline node/chainSimulator/components/testOnlyProcessingNode.go:154]: We need the tx logs processor. Please put back the code

File: node/chainSimulator/process/processor.go [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+import (
+	"github.com/multiversx/mx-chain-core-go/data"
+)
+
+type blocksCreator struct {
+	nodeHandler NodeHandler
+	blsKeyBytes []byte
+}
+
+func NewBlocksCreator(nodeHandler NodeHandler, blsKeyBytes []byte) (*blocksCreator, error) {
```
Comment (iulianpascalau) [inline node/chainSimulator/process/processor.go:12]: missing comment

File: node/chainSimulator/process/processor.go [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+	}, nil
+}
+
+// CreateNewBlock create and process a new block
+func (creator *blocksCreator) CreateNewBlock() error {
+	bp := creator.nodeHandler.GetProcessComponents().BlockProcessor()
+	currentHeader := creator.nodeHandler.GetChainHandler().GetCurrentBlockHeader()
+	var nonce, round uint64
+	var prevHash, prevRandSeed []byte
+	if currentHeader != nil {
```
Comment (iulianpascalau) [inline node/chainSimulator/process/processor.go:25]: please extract in a function L23 - L34

File: node/chainSimulator/process/processor.go [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+	if err != nil {
+		return err
+	}
+
+	err = newHeader.SetPubKeysBitmap([]byte{1})
+	if err != nil {
+		return err
+	}
+
+	err = newHeader.SetChainID([]byte("chain"))
```
Comment (iulianpascalau) [inline node/chainSimulator/process/processor.go:58]: "chain" should be a const and used also in node/chainSimulator/configs/configs.go L51

**Review by iulianpascalau — CHANGES_REQUESTED** [review]

File: node/chainSimulator/components/testOnlyProcessingNode.go (hunk 1 above)
Comment (miiu96) [reply node/chainSimulator/components/testOnlyProcessingNode.go:154]: the logs processor is created inside the process components.

File: node/chainSimulator/process/processor.go (hunk 2 above)
Comment (miiu96) [reply node/chainSimulator/process/processor.go:12]: added

File: node/chainSimulator/process/processor.go (hunk 3 above)
Comment (miiu96) [reply node/chainSimulator/process/processor.go:25]: done

File: node/chainSimulator/process/processor.go (hunk 4 above)
Comment (miiu96) [reply node/chainSimulator/process/processor.go:58]: done

File: node/chainSimulator/components/testOnlyProcessingNode.go (hunk 1 above)
Comment (iulianpascalau) [reply node/chainSimulator/components/testOnlyProcessingNode.go:154]: ok

File: node/chainSimulator/process/processor.go [hunk 5]
```diff
@@ -100,6 +92,21 @@ func (creator *blocksCreator) CreateNewBlock() error {
...
 }
 
+func (creator *blocksCreator) getPreviousHeaderData() (nonce, round uint64, prevHash, prevRandSeed []byte) {
+	currentHeader := creator.nodeHandler.GetChainHandler().GetCurrentBlockHeader()
+
+	if currentHeader != nil {
+		nonce, round = currentHeader.GetNonce(), currentHeader.GetRound()
+		prevHash = creator.nodeHandler.GetChainHandler().GetCurrentBlockHeaderHash()
+		prevRandSeed = currentHeader.GetRandSeed()
+	} else {
```
Comment (iulianpascalau) [inline node/chainSimulator/process/processor.go:102]: could have omitted the "else" branch and return on L101

**Review by iulianpascalau — APPROVED** [review]

File: node/chainSimulator/process/processor.go (hunk 5 above)
Comment (miiu96) [reply node/chainSimulator/process/processor.go]: changed

File: node/chainSimulator/process/processor.go [hunk 6]
```diff
@@ -0,0 +1,181 @@
...
+
+// NewBlocksCreator will create a new instance of blocksCreator
+func NewBlocksCreator(nodeHandler NodeHandler, blsKeyBytes []byte) (*blocksCreator, error) {
+	return &blocksCreator{
+		nodeHandler: nodeHandler,
+		blsKeyBytes: blsKeyBytes,
+	}, nil
+}
+
+// CreateNewBlock create and process a new block
```
Comment (ssd04) [inline node/chainSimulator/process/processor.go:21]:
```suggestion
// CreateNewBlock creates and process a new block
```
?

**Review by ssd04 — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

File: node/chainSimulator/process/processor.go (hunk 6 above)
Comment (iulianpascalau) [reply node/chainSimulator/process/processor.go:21]: fixing in the next PR

**Pull request merged**